src/liboslexec/llvm_util.cpp

```diff
@@ -866,6 +866,11 @@ LLVM_Util::execengine (llvm::ExecutionEngine *exec)
 {
     delete m_llvm_exec;
     m_llvm_exec = exec;
+
+    if (exec == NULL) {
```

Comment:
Maybe a comment here to explain why you need to delete it if and only if exec == nullptr?

Reply:
Turns out argument is always `NULL` - not sure why I thought I needed the if. In any case, should be safe to always delete the profiler.